src/web/client/webViews/QuickPickProvider.ts

```diff
@@ -44,11 +51,51 @@ export class QuickPickProvider {
         } else {
             this.items = [{
                 label: Constants.WEB_EXTENSION_QUICK_PICK_DEFAULT_STRING,
+                id: "",
             }];
         }
     }
 
-    public showQuickPick() {
-        vscode.window.showQuickPick(this.items);
+    public async showQuickPick() {
+        const selectedUser = await vscode.window.showQuickPick(this.items);
+        if (selectedUser) {
+            this.handleSelectedOption(selectedUser);
+        }
+    }
+
+    private handleSelectedOption(selectedOption: IQuickPickItem) {
+        this.collaborationQuickPick(selectedOption);
+    }
+
+    public async collaborationQuickPick(selectedOption: IQuickPickItem) {
+        const collaborationOptions: IQuickPickItem[] = [
+            {
+                label: Constants.START_TEAMS_CHAT,
+                iconPath: {
+                    light: '../assets/microsoftTeamsIcon/light/microsoftTeams.svg',
+                    dark: '../assets/microsoftTeamsIcon/dark/microsoftTeams.svg',
+                }
+            },
+            {
+                label: Constants.SEND_AN_EMAIL,
+                iconPath: new vscode.ThemeIcon(Constants.THEME_ICON_MAIL),
+            },
+        ];
+
+        const collaborationOptionsSelected = await vscode.window.showQuickPick(collaborationOptions, {
+            title: `CONTACT ${selectedOption.label.toUpperCase()}`,
+        });
+
+        if (collaborationOptionsSelected) {
+            if (collaborationOptionsSelected.label === Constants.START_TEAMS_CHAT) {
+                if (selectedOption.id) {
+                    WebExtensionContext.openTeamsChat(selectedOption.id);
+                }
+            } else if (collaborationOptionsSelected.label === Constants.SEND_AN_EMAIL) {
+                if (selectedOption.id) {
+                    WebExtensionContext.openMail(selectedOption.id);
+                }
```

Comment:
you can have separate methods to handle these two separate logic of opening mail and teams chat

Reply:
can you explain this bit more! didn't get what can be done here

Reply:
this is called "Separation of Concern" - separating the logic for handling different collaboration options into distinct methods. this is for better maintainability and modularity.